src/uu/dd/src/dd.rs

```diff
@@ -92,6 +94,20 @@ impl Num {
     }
 }
 
+/// Provide a hint to the kernel that we won't need a portion of a file.
+///
+/// This is a safe version of the [`libc::posix_fadvise`] function
+/// with the advice constant set to [`libc::POSIX_FADV_DONTNEED`].
+#[cfg(any(target_os = "linux", target_os = "android"))]
+fn posix_fadvise_dontneed(fd: i32, offset: i64, len: i64) -> std::io::Result<()> {
```

Comment:
Not super important, but maybe we can use `nix` here to give us a safe wrapper?

Reply:
I'll take a look, thanks.

Reply:
Looks like the `nix` version of `posix_fadvise()` is here: https://docs.rs/nix/latest/nix/fcntl/fn.posix_fadvise.html

Is it worth it to add `nix` as a dependency just for this one function? (Their implementation of this function is more or less the same as this one.) I lean towards "no" but I'll follow your lead. On the other hand, I see that `nix` is already a dependency of many other crates in the repository.

Reply:
Yes, for the multi-call binary it won't matter because it's already a dependency, so I think it's okay. I've been trying to use `nix` over `libc` for new stuff I've written, because then we don't have to bother with `unsafe`. I might make an issue to discuss this as a project-wide direction soon.

Reply:
> Yes, for the multi-call binary it won't matter because it's already a dependency

Good point. Okay then, I'll update this to use the `nix` version of `posix_fadvise`